src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_CAPACITY = 10;
+    private static final double GROW = 1.5;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[START_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            grow();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("IndexOutOfBounds");
+        } else {
+            if (size == array.length) {
+                grow();
+            }
+            T[] newArray = (T[]) new Object[array.length];
+            System.arraycopy(array, 0, newArray,0, index);
+            newArray[index] = value;
+            System.arraycopy(array, index, newArray,index + 1, size - index);
```

Comment:
in each method, use arraycopy only once, you can pass the same array to it twice, as src and as dest